From 58dde58b30658033285155ef99b35b9107220724 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Sat, 18 Mar 2023 11:00:20 +0800
Subject: [PATCH 025/921] [Huawei] md: fix kabi broken in struct mddev

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188282

--------------------------------

Struct mddev is just used inside raid, just in case that md_mod is compiled
from new kernel, and raid1/raid10 or other out-of-tree raid are compiled
from old kernel.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 drivers/md/md.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/md/md.h b/drivers/md/md.h
index 4565f17fc048..0954b9fe4cf6 100644
--- a/drivers/md/md.h
+++ b/drivers/md/md.h
@@ -530,9 +530,9 @@ struct mddev {
 	bool	serialize_policy:1;
 
 	/* Used to synchronize idle and frozen for action_store() */
-	struct mutex sync_mutex;
+	KABI_EXTEND(struct mutex sync_mutex)
 	/* The sequence number for sync thread */
-	atomic_t sync_seq;
+	KABI_EXTEND(atomic_t sync_seq)
 };
 
 enum recovery_flags {
-- 
2.17.1

